src/code/m_field_info.c

```diff
@@ -0,0 +1,428 @@
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008C798_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008C830_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008C84C_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008C8C4_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008C938_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008C964_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008CA14_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008CAD8_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008CC1C_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008CCFC_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008CD24_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008CD74_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008CE00_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008CE38_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008CEC0_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008CFF4_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008D12C_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008D268_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008D3A4_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008D574_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008D6E0_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008D7B0_jp.s")
+
+#pragma GLOBAL_ASM("asm/jp/nonmatchings/code/m_field_info/func_8008D884_jp.s")
+
+s32 mFI_GetWaveUtinBlock(s32* arg0, s32* arg1, s32 arg2, s32 arg3) {
+    UNUSED s32 pad[4];
+    s32 i;
+    s32 ret = 0;
+    s32 var_s2;
+    s32 var_s3 = 0;
+    mCoBG_unkStructUnion* temp_v0;
+    mCoBG_unkStructUnion* var_s1;
+
+    arg0[0] = 0;
+    arg1[0] = 0;
+
+    temp_v0 = var_s1 = func_80089538_jp(arg2, arg3);
+
+    if (temp_v0 != NULL) {
+
+        for (i = 0; i < 0x100; i++) {
+            if (func_80076358_jp(var_s1->data.unk6) == 1) {
+                var_s3 += 1;
+            }
+            var_s1++;
+        }
+
+        if (var_s3 > 0) {
+            var_s1 = temp_v0;
+            var_s2 = (s32)(fqrand() * (f32)var_s3);
+
+            for (i = 0; i < 0x100; i++) {
+
+                if (func_80076358_jp(var_s1->data.unk6) == 1) {
+
+                    if (var_s2 <= 0) {
+                        arg0[0] = i & 0xF;
+                        arg1[0] = i >> 4;
+                        ret = 1;
+                        break;
+
+                    } else {
+                        var_s2 -= 1;
+                    }
+                }
```

Comment:
Since there's a `break` in the previous condition there's no need for an `else`
```suggestion
                if (func_80076358_jp(var_s1->data.unk6) == 1) {

                    if (var_s2 <= 0) {
                        arg0[0] = i & 0xF;
                        arg1[0] = i >> 4;
                        ret = 1;
                        break;
                    }
                    var_s2--;
                }
```